### DISCUSSION TITLE
Mouse input event(s)

### DISCUSSION THREAD

**Discussion by Mikey-Mikey** [wiremod/wire#2773] [Polls]
I'm thinking about doing a pull request to add mouse input events, but I don't know whether to keep them separate or make it an all-in-one event. So I'm asking you guys to help me choose.

**Comment by Denneisk** [comment]
By `mouseMoved`, do you mean an event that outputs the x, y change/position of the cursor/mouse? I could see this being really costly for networking.

I don't see the need for an event for the mouse wheel as the `input` event already tracks that afaik.

**Reply by Mikey-Mikey** [reply]
You don't need to network anything for the mouse event(s). and also the `input` event holds a string as an argument.

**Reply by Mikey-Mikey** [reply]
I've done it in my own core.

**Reply by Mikey-Mikey** [reply]
I also don't even know what the `input` event does because I've tried printing whatever the string is and it shows up nothing.

**Reply by Denneisk** [reply]
> You don't need to network anything for the mouse event(s).

My bad, I was misinformed.

I am also double sorry, for some reason I thought `input` was the `keyPressed` event. The `keyPressed` event outputs the mouse wheel. Of its two string outputs, one is the actual key and the other is the bind, if that key has a binding to it. It outputs `mouse_wheel_up` and `mouse_wheel_down` on the mouse wheel.

**Reply by Mikey-Mikey** [reply]
Yea but it doesn't output how much is scrolled in a single tick. it just outputs that the scrolling happened.
